tests/test_api_validate.py

```diff
@@ -2876,6 +2876,152 @@ def test_00_registrationtoken_with_validity_period(self):
             self.assertEqual("matching 1 tokens, Outside validity period", detail.get("message"))
 
 
+class MultiChallege(MyApiTestCase):
+
+    serial = "hotp1"
+
+    def setUp(self):
+        super(MultiChallege, self).setUp()
+        self.setUp_user_realms()
+
+    def test_00_pin_change_via_validate_chalresp(self):
+        # Test PIN change after challenge response authentication
+        # Create policy change pin on first use
+        set_policy("first_use", scope=SCOPE.ENROLL, action=ACTION.CHANGE_PIN_FIRST_USE)
+        set_policy("via_validate", scope=SCOPE.AUTH, action=ACTION.CHANGE_PIN_VIA_VALIDATE)
+        set_policy("hotp_chalresp", scope=SCOPE.AUTH, action="{0!s}=hotp".format(ACTION.CHALLENGERESPONSE))
+
+        with self.app.test_request_context('/token/init', method='POST',
+                                           data={"user": "cornelius", "pin": "test",
+                                                 "serial": self.serial, "otpkey": self.otpkey},
+                                           headers={"Authorization": self.at}):
+            res = self.app.full_dispatch_request()
+            self.assertEqual(res.status_code, 200)
+            result = res.json['result']
+            self.assertTrue(result.get("value"))
+
+        # 1st authentication creates a PIN change challenge via challenge response
+        with self.app.test_request_context('/validate/check', method='POST',
+                                           data={"user": "cornelius",
+                                                 "pass": "test"}):
+            res = self.app.full_dispatch_request()
+            self.assertEqual(res.status_code, 200)
+            details = res.json['detail']
```

Comment:
Please also check that the result value is `False` so we don't miss any possible mis-configurations

Reply:
done in f7589747